public/components/metrics/redux/slices/metrics_slice.ts

```diff
@@ -52,25 +55,34 @@ const initialState = {
     recentlyUsedRanges: [],
   },
   refresh: 0, // set to new Date() to trigger
+  selectedDataSource: '',
+  otelIndices: [],
+  otelDocumentNames: [],
 };
 
 export const loadMetrics = () => async (dispatch) => {
   const { http, pplService } = coreRefs;
   const customDataRequest = fetchCustomMetrics(http);
   const remoteDataSourcesResponse = await pplServiceRequestor(pplService!, PPL_DATASOURCES_REQUEST);
   const remoteDataSources = remoteDataSourcesResponse.data.DATASOURCE_NAME;
-
   dispatch(setDataSources(remoteDataSources));
   dispatch(setDataSourceTitles(remoteDataSources));
   dispatch(
-    setDataSourceIcons(coloredIconsFrom([OBSERVABILITY_CUSTOM_METRIC, ...remoteDataSources]))
+    setDataSourceIcons(
+      coloredIconsFrom([OBSERVABILITY_CUSTOM_METRIC, ...remoteDataSources, 'OpenTelemetry'])
+    )
   );
 
   const remoteDataRequests = await fetchRemoteMetrics(remoteDataSources);
   const dataResponses = await Promise.all([customDataRequest, ...remoteDataRequests]);
   dispatch(setMetrics(dataResponses.flat()));
 };
 
+export const loadOTIndices = () => async (dispatch) => {
```

Comment:
Why do you want this separate from `loadMetrics`?  Seems like it should just be integrated?

Reply:
the mechanism is to either load promql metrics or otel metrics in available metrics, but if i integrate loading `loadOTIndices` as a part of `loadMetrics`, then otel indices won't be loaded even if the datasource is selected as OpenTelemetry. Hence its separate